Source/Core/Core/IOS/USB/Emulated/Skylanders/SkylanderFigure.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <Common/IOFile.h>
+
+#include <span>
+
+namespace IOS::HLE::USB
+{
+struct SkylanderDateTime final
+{
+  u8 minute;
+  u8 hour;
+  u8 day;
+  u8 month;
+  u16 year;
+};
+
+struct SkylanderData final
+{
+  std::array<u8, 11> toy_code;
+  u16 money;
+  u16 hero_level;
+  u32 playtime;
+  // Null-terminated UTF-16 string
+  std::array<u16, 0x10> nickname;
+  SkylanderDateTime last_reset;
+  SkylanderDateTime last_placed;
+};
+
+struct Trophydata final
+{
+  u8 unlocked_villains;
+};
+
+struct FigureData final
+{
+  u8 normalized_type;
```

Comment:
To do this, I needed to move the Type enum to the SkylanderFigure.h file to avoid header files including each other. I don't really like this because it splits the enums for SkyData over different files. Any recommendations on how to handle this?

Reply:
You can forward declare `enum class`es.